New test_fileobj.py file now fails on MacOS 10.15

This is a new thing for me. I now get:

```
tests/test_fileobj.py:290: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/anaconda3/lib/python3.8/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
/usr/local/anaconda3/lib/python3.8/multiprocessing/context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
/usr/local/anaconda3/lib/python3.8/multiprocessing/context.py:283: in _Popen
    return Popen(process_obj)
/usr/local/anaconda3/lib/python3.8/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
/usr/local/anaconda3/lib/python3.8/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
/usr/local/anaconda3/lib/python3.8/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = <Process name='Process-1' parent=74259 initial>, file = <_io.BytesIO object at 0x7f9e5e981810>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'TestSocket.client_fd.<locals>.server_process'

/usr/local/anaconda3/lib/python3.8/multiprocessing/reduction.py:60: AttributeError
============================================================== short test summary info ===============================================================
ERROR tests/test_fileobj.py::TestSocket::test_single_then_multi - AttributeError: Can't pickle local object 'TestSocket.client_fd.<locals>.server_p...
```

--- a/python/tests/test_fileobj.py
+++ b/python/tests/test_fileobj.py
@@ -260,30 +260,33 @@ class TestFIFO:
 ADDRESS = ("localhost", 10009)
 
 
+class Server(socketserver.ThreadingTCPServer):
+    allow_reuse_address = True
+
+
+class StoreEchoHandler(socketserver.BaseRequestHandler):
+    def handle(self):
+        while True:
+            try:
+                ts = tskit.load(self.request.fileno())
+            except EOFError:
+                break
+            ts.dump(self.request.fileno())
+        self.server.shutdown()
+
+
+def server_process(q):
+    server = Server(ADDRESS, StoreEchoHandler)
+    # Tell the client (on the other end of the queue) that it's OK to open
+    # a connection
+    q.put(None)
+    server.serve_forever()
+
+
 @pytest.mark.skipif(IS_WINDOWS, reason="Errors on Windows")
 class TestSocket:
     @fixture
     def client_fd(self):
-        class Server(socketserver.ThreadingTCPServer):
-            allow_reuse_address = True
-
-        class StoreEchoHandler(socketserver.BaseRequestHandler):
-            def handle(self):
-                while True:
-                    try:
-                        ts = tskit.load(self.request.fileno())
-                    except EOFError:
-                        break
-                    ts.dump(self.request.fileno())
-                self.server.shutdown()
-
-        def server_process(q):
-            server = Server(ADDRESS, StoreEchoHandler)
-            # Tell the client (on the other end of the queue) that it's OK to open
-            # a connection
-            q.put(None)
-            server.serve_forever()
-
         # Use a queue to synchronise the startup of the server and the client.
         q = multiprocessing.Queue()
         _server_process = multiprocessing.Process(target=server_process, args=(q,))

